pkg/virt-controller/watch/vm/vm_test.go

```diff
@@ -5401,6 +5411,44 @@ var _ = Describe("VirtualMachine", func() {
 					Expect(cond.Status).To(Equal(k8sv1.ConditionTrue))
 					Expect(cond.Message).To(ContainSubstring("invalid volumes to update with migration:"))
 				})
+
+				It("should return an error when the destination PVC doesn't exist", func() {
+					testutils.UpdateFakeKubeVirtClusterConfig(kvStore, &v1.KubeVirt{
+						Spec: v1.KubeVirtSpec{
+							Configuration: v1.KubeVirtConfiguration{
+								VMRolloutStrategy: &liveUpdate,
+							},
+						},
+					})
+					dv := libdv.NewDataVolume(libdv.WithName(newDVName), libdv.WithNamespace(ns))
+					Expect(dataVolumeInformer.GetStore().Add(dv)).To(Succeed())
+					vmi := libvmi.New(libvmi.WithNamespace(ns), libvmi.WithDataVolume(diskName, oldDVName))
+					vm := libvmi.NewVirtualMachine(libvmi.New(libvmi.WithNamespace(ns),
+						libvmi.WithDataVolume(diskName, newDVName)),
+						libvmi.WithUpdateVolumeStrategy(v1.UpdateVolumesStrategyMigration))
+					err := controller.handleVolumeUpdateRequest(vm, vmi)
+					Expect(err).To(MatchError(storagetypes.NewPVCNotFoundError(newDVName)))
+					Expect(virtcontroller.NewVirtualMachineConditionManager().GetCondition(vm,
+						v1.VirtualMachineRestartRequired)).Should(BeNil())
+				})
+
+				It("should return an error when the destination DV doesn't exist", func() {
+					testutils.UpdateFakeKubeVirtClusterConfig(kvStore, &v1.KubeVirt{
+						Spec: v1.KubeVirtSpec{
+							Configuration: v1.KubeVirtConfiguration{
+								VMRolloutStrategy: &liveUpdate,
+							},
+						},
+					})
+					vmi := libvmi.New(libvmi.WithNamespace(ns), libvmi.WithDataVolume(diskName, oldDVName))
+					vm := libvmi.NewVirtualMachine(libvmi.New(libvmi.WithNamespace(ns),
+						libvmi.WithDataVolume(diskName, newDVName)),
+						libvmi.WithUpdateVolumeStrategy(v1.UpdateVolumesStrategyMigration))
+					err := controller.handleVolumeUpdateRequest(vm, vmi)
+					Expect(err).To(MatchError(storagetypes.NewDVNotFoundError(newDVName)))
+					Expect(virtcontroller.NewVirtualMachineConditionManager().GetCondition(vm,
+						v1.VirtualMachineRestartRequired)).Should(BeNil())
+				})
```

Comment:
Looks like these can be a `DescribeTable`

Reply:
I thought about that, but I had to introduce a bool to check if it was a dv or pvc, so I thougth to keep to separate for this reason, but I can merge them